### PR TITLE
added post_type to notify::notify_mentions

### LINKED ISSUE #73: Notifications should pass the post type
This is related to [#505](https://github.com/near/neardevhub-widgets/issues/505) and [#504](https://github.com/near/neardevhub-widgets/issues/504) in neardevhub-widgets repository. The goal is to ensure that the notification links route correctly based on the `postType`.

In the case of a blog, the postType is a "comment" with no parentId... this was a hack to get Blogs out fast. We may also want to explore formally creating a "Blog" type and migrating existing "blogs" to this type.

### Implementation:

1.  Modify  [`notify.rs`](https://github.com/near/neardevhub-contract/blob/bbadebda949d98cdb23e1f2bda22fc625b51a79e/src/notify.rs#L42) to ensure that the `postType` is included alongside the ID in function parameters.

2. Update notify functions calls to pass the post type (comes from post body) 
     - [Reply Notification](https://github.com/near/neardevhub-contract/blob/bbadebda949d98cdb23e1f2bda22fc625b51a79e/src/lib.rs#L189)
     - [Edit Notification](https://github.com/near/neardevhub-contract/blob/bbadebda949d98cdb23e1f2bda22fc625b51a79e/src/lib.rs#L328)
     - [Like Notification](https://github.com/near/neardevhub-contract/blob/bbadebda949d98cdb23e1f2bda22fc625b51a79e/src/lib.rs#L133)
     - [Mention Notification](https://github.com/near/neardevhub-contract/blob/bbadebda949d98cdb23e1f2bda22fc625b51a79e/src/lib.rs#L193)

3. Either resolve "comment" type with no parentId as "blog", or create a blog type and migrate existing posts.

### PR DESCRIPTION
**Pull request by Tguntenaar**
_Resolves #73_
and unblocks [#505](https://github.com/NEAR-DevHub/neardevhub-bos/issues/505)

The notify module now passes the post_type to tell what type of post triggered the notification.

As Elliot mentioned [here](https://github.com/NEAR-DevHub/neardevhub-contract/issues/73#issue-2017757824) we resolve "comment" type with no parentId as "blog". 

I first added a blog as a type for a post but it would imply a new PostBodyV1 a new PostSnapshotV1 and a whole bunch of extra code to migrate without a real gain. Didn't want to overengineer it.

### CODE REVIEW

File: src/notify.rs [hunk 1]
```diff
@@ -78,8 +79,11 @@ fn notify(post_id: PostId, post_author: AccountId, action: &str) -> Promise {
                     "notify": json!({
                         "key": post_author,
                         "value": {
+                            // the custom types
+                            // action is edit reply like or mention
                             "type": format!("devgovgigs/{}", action),
                             "post": post_id,
+                            "post_type": post_type
```
Comment (Tguntenaar) [inline src/notify.rs:84]: @elliotBraem is this what what you meant in the issue?

**Comment by elliotBraem** [comment]
@frol @ailisp this PR looks good to me and resolves the change as I intended it. Could someone review and merge please to unblock the front end PR?

**Review by Tguntenaar — COMMENTED** [review]
@frol @ailisp this is ready for your review

**Comment by frol** [comment]
I am sorry to jump on it late, but I feel it is pointless to fix this functionality given that we plan to implement announcements, discussions, and proposals using SocialDB posts and thus will gain all the notifications out of the box.

This implementation looks pretty weird to me from the design point of view, but instead of trying to clean it up, I would just drop it.

**Comment by Tguntenaar** [comment]
Alright will close this PR since we are dropping it

**Pull request closed without merging**